### PR TITLE
Add variable type selector to optimize API usage

### PR DESCRIPTION
**Pull request by Loori-R**
add variable type selector to optimize API usage, favoring [/api/v1/labels](https://docs.victoriametrics.com/url-examples/#apiv1labels) and [/api/v1/label/.../values](https://docs.victoriametrics.com/url-examples/#apiv1labelvalues) over [/api/v1/series](https://docs.victoriametrics.com/url-examples/#apiv1series). 

#144

image

### CODE REVIEW

File: src/language_provider.ts [hunk 1]
```diff
@@ -483,6 +485,49 @@ export default class PromQlLanguageProvider extends LanguageProvider {
...
+    const range = this.datasource.getAdjustedInterval(this.timeRange);
+    const urlParams = {
+      ...range,
+      'match[]': interpolatedName,
+    };
+    const url = `/api/v1/labels`;
+
+    const data: string[] = await this.request(url, [], urlParams);
+    // Convert string array to Record<string , []>
+    return data.reduce((ac, a) => ({ ...ac, [a]: '' }), {});
```
Comment (dmitryk-dk) [inline src/language_provider.ts:511]: from the comment i see that the `Record` should be `Record<string , []>`, but in the reduce the object is `{[string]: string}`. Maybe I miss something

File: src/metric_find_query.ts [hunk 2]
```diff
@@ -16,41 +16,63 @@
...
+      );
+    }
+
     if (labelNamesQuery) {
-      return this.labelNamesQuery();
+      return this.datasource.getTagKeys({ filters: [], timeRange });
     }
 
     const labelValuesQuery = this.query.match(labelValuesRegex);
     if (labelValuesQuery) {
```
Comment (dmitryk-dk) [inline src/metric_find_query.ts:68]: i think we should check the length also

**Review by dmitryk-dk — COMMENTED** [review]
Overall looks good! Add some somments

File: src/metric_find_query.ts (hunk 2 above)
Comment (Loori-R) [reply src/metric_find_query.ts:68]: In case of no matches, the result will be `null`, not an empty array.

File: src/language_provider.ts (hunk 1 above)
Comment (Loori-R) [reply src/language_provider.ts]: Yes, there's a mistake here. I checked, only keys are used everywhere, so it's more logical to leave it as `string[]`.

**Review by dmitryk-dk — DISMISSED** [review]
LGTM!

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

**Pull request merged**